crux_macros/src/effect.rs

```diff
@@ -23,11 +23,18 @@ impl ToTokens for EffectStructReceiver {
     fn to_tokens(&self, tokens: &mut TokenStream) {
         let ident = &self.ident;
 
-        let effect_name = match self.name {
-            Some(ref name) => quote!(#name),
+        let (effect_name, ffi_effect_name) = match self.name {
+            Some(ref name) => {
+                let ef_name = name;
+                let ffi_ef_name = format_ident!("{}Ffi", name);
+
+                (quote!(#ef_name), quote!(#ffi_ef_name))
+            }
             None => {
-                let x = Type::from_string("Effect").unwrap();
-                quote!(#x)
+                let ef_name = Type::from_string("Effect").unwrap();
+                let ffi_ef_name = Type::from_string("EffectFfi").unwrap();
+
+                (quote!(#ef_name), quote!(#ffi_ef_name))
```

Comment:
I may be missing something, but couldn't this be 
```suggestion
                (quote!(Effect), quote!(EffectFfi))
```